pkg/service/restore/schema_worker.go

```diff
@@ -123,6 +132,47 @@ func (w *schemaWorker) stageRestoreData(ctx context.Context) error {
 	return nil
 }
 
+func (w *schemaWorker) restoreFromSchemaFile(ctx context.Context) error {
+	w.logger.Info(ctx, "Apply schema CQL statements")
+	start := timeutc.Now()
+	for _, row := range *w.describedSchema {
+		if row.Keyspace == "system_replicated_keys" {
+			// See https://github.com/scylladb/scylla-enterprise/issues/4168
+			continue
+		}
+		// Sometimes a single object might require multiple CQL statements (e.g. table with dropped and added column)
+		for _, stmt := range parseCQLStatement(row.CQLStmt) {
+			if err := w.clusterSession.ExecStmt(stmt); err != nil {
+				return errors.Wrapf(err, "create %s (%s) with %s", row.Name, row.Keyspace, stmt)
+			}
```

Comment:
I think that it may lead to inconsistent behavior when some of the statements succeeds, but it fail on the next one.
First run will end on first incorrect statement, the second one will error on the first statement as the table may exist already.
WDYT about dropping created keyspaces on failed run ? Kind of rollback.

Reply:
We can do that!
First I wanted to say that it would require us to parse CQL statements (bad idea), so that we would know what should be dropped, but thanks to the new schema file format, we save information about all keyspace names present in restored schema, so it makes it simple and safe to do!